runtime/decimal.c

```diff
@@ -129,6 +131,37 @@ TaggedPtr _bal_decimal_neg(TaggedPtr tp) {
     return createDecimal(&d);
 }
 
+bool _bal_decimal_exact_eq(TaggedPtr tp1, TaggedPtr tp2) {
+    decQuad d1;
+    decQuad d2;
+    decQuadCanonical(&d1, taggedToDecQuad(tp1));
+    decQuadCanonical(&d2, taggedToDecQuad(tp2));
+    return memcmp(&d1, &d2, DECQUAD_Bytes) == 0;
+}
+
+int64_t _bal_decimal_cmp(TaggedPtr tp1, TaggedPtr tp2) {
+    decQuad d;
+    decContext cx;
+    initContext(&cx);  
+    decQuadCompare(&d, taggedToDecQuad(tp1), taggedToDecQuad(tp2), &cx);
+    // The return value of decQuadCompare can be 1, 0, -1 or NaN.
+    // NaN is returned only if lhs or rhs is a NaN.
+    // In _bal_decimal_cmp, both arguments are valid decimals.
+    // Therefore NaN is not an expected output.
+    // Since the result is a decQuad, decQuadClass is used to
```

Comment:
Move the last two lines of the comment to be under the first line and combind

```
The return value of decQuadCompare can be 1, 0, -1 or NaN:
since this is represnted as a decQuad, decQuadClass is used to
identify 1, 0, -1.
```

Reply:
Changed